mongo_test.go

```diff
@@ -229,13 +242,82 @@ func TestClear(t *testing.T) {
 		assert.NoError(t, err)
 		assert.Equal(t, 2, deleted)
 	}
+	assertCollectionIDs(t, s.DB(dbName).C(dbName), []string{"1", "2"})
 
 	q, err = query.New("", `{id:"2"}`, "", nil)
 	if assert.NoError(t, err) {
 		deleted, err := h.Clear(context.Background(), q)
 		assert.NoError(t, err)
 		assert.Equal(t, 1, deleted)
 	}
+	assertCollectionIDs(t, s.DB(dbName).C(dbName), []string{"1"})
+}
+func TestClearLimit(t *testing.T) {
```

Comment:
The update of the tests is done in a lazy way atm. Ideally we would rewrite this file to drop the testify dependency as well as use table tests for less code/check duplication. However, I did not want to spend any time on that for this PR.